include/ddc/chunk.hpp

```diff
@@ -60,23 +61,32 @@ class Chunk<ElementType, DiscreteDomain<DDims...>>
 
     using reference = typename base_type::reference;
 
-    template <class, class>
+    template <class, class, class>
     friend class Chunk;
 
+private:
+    Allocator m_allocator;
+
 public:
     /// Empty Chunk
     Chunk() = default;
 
     /// Construct a Chunk on a domain with uninitialized values
-    explicit Chunk(mdomain_type const& domain)
-        : base_type(new (std::align_val_t(64)) value_type[domain.size()], domain)
+    explicit Chunk(mdomain_type const& domain, Allocator const& allocator = Allocator())
+        : base_type()
+        , m_allocator(allocator)
     {
+        static_cast<base_type&>(*this) = base_type(
+                std::allocator_traits<Allocator>::allocate(m_allocator, domain.size()),
+                domain);
```

Comment:
Why not do it in the initializer list like before?

Reply:
Because the `base_type` constructor will be called first, using the uninitialised member `m_allocator`, and then initialize `m_allocator`. I agree it is not pretty.

Reply:
Why not just
```
: base_type(std::allocator_traits<Allocator>::allocate(allocator, domain.size()), domain)
, m_allocator(allocator)
```

is it an issue to use the allocator that will be copied instead of the actual member?

Reply:
otherwise, maybe add a protected function dedicated to the allocation in `ChunkCommon`, this could be more explicit than this strange use of the move assignment